Src/WitsmlExplorer.Frontend/components/Sidebar/ServerManager.tsx

```diff
@@ -39,7 +41,7 @@ const ServerManager = (): React.ReactElement => {
 
   useEffect(() => {
     const onCurrentLoginStateChange = async () => {
-      if (currentLoginState.server) {
+      if (!msalEnabled && currentWitsmlLoginState.server) {
```

Comment:
Shouldn't the !msalEnabled check be done outside the if? As in:
if(!msalEnabled){
  if(currentWitsmlLoginState.server){
   ...

Reply:
Possibly invert the logic as well to check for msalEnabled instead of the negation

Reply:
Thanks, I will look into it. There is a logical flaw aswell causing this to trigger before we have the token needed to fetch wells.

Reply:
Updated...